R/getPath.R

```diff
@@ -16,8 +11,10 @@ getScenNamesFast <- function(outputDirs) {
 #' @rdname getPath
 #' @export
 getMifScenPath <- function(outputDirs, mustWork = FALSE) {
-  names <- getScenNamesFast(outputDirs)
-  path <- file.path(outputDirs, paste0("REMIND_generic_", names, ".mif"))
+  # Find all mif files starting with REMIND_generic_. The result will also contain REMIND_generic_withoutPlus.mif
+  reports <- list.files(outputDirs, "^REMIND_generic_.*\\.mif$", full.names = TRUE)
+  # Filter out filenames having "withoutPlus.mif" in their name
+  path <- reports[grepl("REMIND_generic_(?!.*(withoutPlus)\\.mif).*\\.mif$", reports, perl = TRUE)]
```

Comment:
```suggestion
  path <- reports[grepl("^REMIND_generic_(?!.*(withoutPlus)\\.mif).*\\.mif$", reports, perl = TRUE)]
```
Or is there a reason to avoid `^` here? For me, it is always easier to think about regexen with `^` and `$`, because I don't have to think about the effects of the implicit `.*` at the start or end, respectively.

Reply:
Also, couldn't there be multiple mifs? Then, `path` might be a vector with length > 1, and everything explodes?

Reply:
Hmmm… if you do it in two steps, you can also radically simplify the `grep`, I think:

```R
path <- grep("withoutPlus\\.mif$", reports, invert = TRUE)
```
does the job, right?

Reply:
Argh, you also need a `value = TRUE`:
```R
path <- grep("withoutPlus\\.mif$", reports, invert = TRUE, value = TRUE)
```

Reply:
Note that further .mif files could be present, such as `_adjustedPolicyCosts.mif`. I think it would be safer to use https://github.com/pik-piam/lucode2/blob/master/R/getScenNames.R

Reply:
> Or is there a reason to avoid `^` here? For me, it is always easier to think about regexen with `^` and `$`, because I don't have to think about the effects of the implicit `.*` at the start or end, respectively.

Yes. Since `list.files(..., full.names = TRUE)` returns the full paths (not only the file name), the string does not start with "REMIND...".

Reply:
> Hmmm… if you do it in two steps, you can also radically simplify the `grep`, I think:
> 
> ```r
> path <- grep("withoutPlus\\.mif$", reports, invert = TRUE)
> ```
> 
> does the job, right?

True. Thanks. Btw: didn't know the `invert` argument :-)

Reply:
> Note that further .mif files could be present, such as `_adjustedPolicyCosts.mif`. I think it would be safer to use https://github.com/pik-piam/lucode2/blob/master/R/getScenNames.R

Ok, if this is the case, we should indeed use getScenNames.R I wanted to avoid this because to me it seems a bit more time-consuming, but maybe not relevant here.